File: problems/字符串/151.反转字符串中的单词.py
#!/usr/bin/env python3
"""
151. 反转字符串中的单词
"""


def reverse_words(s: str) -> str:
    return ' '.join(s.strip().split()[::-1])


if __name__ == "__main__":
    # 测试用例
    print(reverse_words("the sky is blue"))      # 预期: "blue is sky the"
    print(reverse_words("  hello world  "))     # 预期: "world hello"
    print(reverse_words("a good   example"))    # 预期: "example good a"

File: problems/字符串/151.反转字符串中的单词.cpp
#include <iostream>
#include <sstream>
#include <string>
using namespace std;

string reverseWords(string s) {
    istringstream iss(s);
    string word, result;
    while (iss >> word) {
        if (!result.empty()) result += ' ';
        result = word + (result.empty() ? "" : " " + result);
    }
    return result;
}

int main() {
    // 测试用例
    cout << reverseWords("the sky is blue") << endl;      // 预期: blue is sky the
    cout << reverseWords("  hello world  ") << endl;     // 预期: world hello
    cout << reverseWords("a good   example") << endl;    // 预期: example good a
    return 0;
}
